apps/aeutils/src/aeu_requests.erl

```diff
@@ -27,6 +27,19 @@ ping(Peer) ->
     PingObj = PingObj0#{<<"peers">> => Peers},
     Response = process_request(Peer, post, Req, PingObj),
     case Response of
+        {ok, #{<<"reason">> := Reason}} ->
+            lager:debug("Got an error return: Reason = ~p", [Reason]),
+            case Reason of
+                <<"Different genesis", _/binary>> ->
+                    aec_peers:block_peer(Uri);
+                <<"Not allowed", _/binary>> ->
+                    aec_peers:block_peer(Uri);
+                _ -> ok
+            end,
+            aec_events:publish(chain_sync,
+                               {sync_aborted, #{uri => Uri,
+                                                reason => Reason}}),
+            {error, Reason};
         {ok, Map} ->
             lager:debug("ping response (~p): ~p", [Uri, pp(Map)]),
             aec_sync:compare_ping_objects(PingObj, Map),
```

Comment:
I am a bit surprised that `{error, different_genesis_blocks}` is not checked here. Maybe a comment here would clarify for reader?

----

> A peer is auto-blocked if it passes the wrong genesis block in a ping

I see this is from the point of view of the ping server. Ok.

> A pinging peer also blocks the other peer if it receives the ping response "Different genesis blocks" or "Not allowed"

I see this is from the point of view of the ping client. Ok.

----

Also it is not clear to me whether when node connects to peer, the peer establishes another connection back.

Reply:
The different genesis blocks outcome is checked on [line 33](https://github.com/aeternity/epoch/blob/822ad0f3be7709cc63f0986772d399310d69510b/apps/aeutils/src/aeu_requests.erl#L33) in aeu_requests. The return values from `process_request()` are a bit misleading.

When a node pings a peer, it's via a HTTP request, so the peer responds with a `PingObject`. Depending of the outcome of the PingObject comparisons, one node may designate itself as client and proceed to fetch blocks and headers from the other node – also via HTTP requests. Both nodes will request the mempool from the other node, and both will request missing blocks, so there will be HTTP requests in both directions.